File: solutions/vbnet/resistor-color-duo/1/ResistorColorDuo.vb
Imports System

Public Module ResistorColorDuo
    Public Function Value(ByVal colors As String()) As Integer
        Dim colorList As String() = New String() {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}
        Return Integer.Parse(Array.IndexOf(colorList, colors(0)).ToString() & Array.IndexOf(colorList, colors(1)).ToString())
    End Function
End Module

File: solutions/r/resistor-color-duo/1/resistor-color-duo.R
value <- function(resistor_colors) {
  color_values <- c(
    black = 0,
    brown = 1,
    red = 2,
    orange = 3,
    yellow = 4,
    green = 5,
    blue = 6,
    violet = 7,
    grey = 8,
    white = 9
  )

  digits <- unname(color_values[resistor_colors[1:2]])

  digits[1] * 10 + digits[2]
}